crates/neon/src/types_impl/promise.rs

```diff
@@ -49,9 +49,74 @@ const BOUNDARY: FailureBoundary = FailureBoundary {
     feature = "promise-api",
     deprecated = "`promise-api` feature has no impact and may be removed"
 )]
-/// The JavaScript [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) value.
+/// The type of JavaScript
+/// [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+/// objects.
 ///
-/// [`JsPromise`] may be constructed with [`Context::promise`].
+/// [`JsPromise`] instances may be constructed with [`Context::promise`].
```

Comment:
Is it worth explaining the role of `Deferred` here? We could include the equivalent JavaScript code for reference:

```js
let deferred;
let promise = new Promise((resolve, reject) => deferred = { resolve, reject });

deferred.resolve("Hello, World!");
```

Reply:
great idea, added this